app/tests/conftest.py

```
@@ -208,6 +208,26 @@ def update_password(self, user_id: str, password_hash: str):
         return updated
 
+    def get_by_provisioning_token(self, token: str):
+        for user in self._by_id.values():
+            if user.provisioning_token == token:
+                return user
+        return None
+
+    def set_provisioning_token(self, user_id: str, token: str) -> None:
+        user = self.get_by_id(user_id)
+        if not user:
+            raise ValueError(f"User {user_id} not found")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)